homeassistant/components/motionmount/__init__.py

```diff
@@ -0,0 +1,55 @@
+"""The Vogel's MotionMount integration."""
+from __future__ import annotations
+
+import socket
+
+import motionmount
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+from .coordinator import MotionMountCoordinator
+
+PLATFORMS: list[Platform] = [
+    Platform.NUMBER,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Vogel's MotionMount from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
+
+    # Create API instance
+    mm = motionmount.MotionMount(entry.data[CONF_HOST], entry.data[CONF_PORT])
+    coordinator = MotionMountCoordinator(hass, mm)
+    mm.add_listener(coordinator.motionmount_callback)
+
+    # Validate the API connection
+    try:
+        await mm.connect()
+    except (ConnectionError, TimeoutError, socket.gaierror) as ex:
+        raise ConfigEntryNotReady(
+            f"Failed to connect to {entry.data[CONF_HOST]}"
+        ) from ex
+    await coordinator.async_config_entry_first_refresh()
+
+    # Store an API object for your platforms to access
```

Comment:
We need to abort if the unique id does not match the one in the config entry as otherwise the ip could have changed and we are now talking to the wrong device


There is an example of how to check this in tplink, flux_led, and lifx